modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -228,15 +250,23 @@ public static void checkInstantiable(Class<?> declaringClass)
 
     public static void checkInstantiable(Class<?> declaringClass, boolean requireDefaultConstructor)
     {
-        checkArgument(declaringClass != null, "declaringClass cannot be null");
-
-        if (requireDefaultConstructor)
+        if (!isInstantiable(declaringClass, requireDefaultConstructor))
         {
-            checkArgument(hasDefaultConstructor(declaringClass), String.format("Class %s cannot be instantiated since it doesn't have a default constructor", declaringClass.getName()));
+            checkArgument(false, String.format("Class %s cannot be instantiated.", declaringClass));
```

Comment:
checkARgument(false) doesn't make much sense... Simply throw the IllegalArgumentException yourself